Okta/AppAuth/OIDAuthorizationService.m

```diff
@@ -420,18 +420,19 @@ + (void)discoverServiceConfigurationForDiscoveryURL:(NSURL *)discoveryURL
 
 #pragma mark - Token Endpoint
 
-+ (void)performTokenRequest:(OIDTokenRequest *)request callback:(OIDTokenCallback)callback {
-  [[self class] performTokenRequest:request
-      originalAuthorizationResponse:nil
-                           callback:callback];
++ (void)performTokenRequest:(OIDTokenRequest *)request
+                   delegate:(id<OktaOidcHTTPProtocol> _Nullable)delegate
+                   callback:(OIDTokenCallback)callback {
+    [[self class] performTokenRequest:request originalAuthorizationResponse:nil delegate:delegate callback:callback];
 }
 
 + (void)performTokenRequest:(OIDTokenRequest *)request
-    originalAuthorizationResponse:(OIDAuthorizationResponse *_Nullable)authorizationResponse
-                         callback:(OIDTokenCallback)callback {
+originalAuthorizationResponse:(OIDAuthorizationResponse *_Nullable)authorizationResponse
+                   delegate:(id<OktaOidcHTTPProtocol> _Nullable)delegate
+                   callback:(OIDTokenCallback)callback {
 
   NSURLRequest *URLRequest = [request URLRequest];
-  
+  [delegate willSendRequest:URLRequest];
```

Comment:
Check for protocol conformance

Reply:
@IldarAbdullin-okta  and @lihaoli-okta  Another query I have is that if the customer wants to modify the request, how can they modify it. This request is not mutable.

One more suggestion I have is that, instead of using the name willSendRequest, we should rename it to something that defines the purpose of the call. The purpose is to give back a modified version URLRequest and it should explicitly return modified version of the request. Mutating a passed parmeter is not a recommended pattern.

Reply:
If they want to modify the request, they can create new one based on provided URLRequest. I really don't to introduce more changes in swift/core parts for that very optional feature